svelte/src/scripts/create-graph-data/index.ts

```diff
@@ -69,25 +96,36 @@ export function createGraphData(convertedData: ConvertedData): GraphData {
 
 	assignParentReference(graphDataNodes);
 
-	// add originalIncoming and outgoingLinks
+	// setup to be assigned
 	graphDataFlattenNodes.forEach((node) => {
+		// all are reference reference
 		node.outgoingLinks = [];
 		node.originalOutgoingLinks = [];
 		node.incomingLinks = [];
 		node.originalIncomingLinks = [];
+		node.outgoingLinksLifted = [];
+		node.incomingLinksLifted = [];
 	});
 
-	const graphDataLinks = assignOutgoingAndIncomingLinksAndOriginalSourceAndTargetReference(
+	const graphDataLinks = assignOutgoingIncomingLinksAndOriginalLiftedSourceTargetReference(
 		links,
 		flattenNodes,
 		graphDataFlattenNodes
 	);
 
+	// create nodesDict
+	const nodesDict: { [id: string]: GraphDataNode } = {};
+	graphDataFlattenNodes.forEach((node) => {
```

Comment:
Since Satrio took issue with the flattenNodes and the nodesDict specifically, maybe the GraphData should be wrapped in a class? Then those interfaces could be a façade. Though that may be for another PR.